oeps/processes/oep-0055-proc-project-maintainers.rst

```diff
@@ -0,0 +1,428 @@
+acceptable for the team to figure out arrangements for list monitoring, such as
+an "on-call" rotation.
+
+Prerequisites
+-------------
+In order to be a repository maintainer, you must
+
+* Be an active member of the Open edX community who has made technical
+  contributions to the project in the last 6 months.  The types of
+  contributions that are relevant will match those defined in the
+  `code contributors`_
+* Abide by the Open edX `Code of Conduct`_
+  <https://openedx.org/code-of-conduct/>`_
+* Be a member of the `Core Contributor Program`_ with rights to the repository
+  in question who has demonstrated excellence in the role of core contributor.
+  From time-to-time, in the case of a strong candidate and project need, core
+  contributor status and maintainer status may be conferred at the same time.
+* In the case of repos being moved into the `openedx GitHub organization`_, be
+  an original maintainer of the code.
+
+Bootstrapping
+-------------
+In order to ensure an orderly roll-out of the formal maintainer role,
+existing 2U staff in the owner role will transition into the role of
+maintainer for the components that they own.
+
+Community members who are acting in informal maintainer roles will
+also be asked to become named maintainers.
+
+A list of all components without maintainers will be created and
+published.
+
+An asynchronous channel will be available for all maintainers to
+communicate and collaborate; initially this will be a Slack channel.
+
+Standards for measuring the quality of repositories will be rolled out
+across all repositories.  These standards can benefit from work done
+at edX to measure `repository health <https://github.com/openedx/edx-repo-health>`_.
+
+The Maintainer Scrum-of-Scrums
+------------------------------
+In order to coordinate certain maintainer responsibilities including
+voting on nominated maintainers, iterating on standards, and removing
+maintainers, minimal mechanisms for coordination are required.
+
+For now, we will assume that maintainers will self-organize to
+complete the critical activities.
+
+Coordination will typically be over asynchronous channels.
+
+From time-to-time, a task force may be required to complete certain
+project-wide responsibilities like developing repository standards,
+as well as tools for measuring compliance to those standards.
+
+Becoming a Maintainer
+---------------------
+After the bootstrapping phase, the Open edX project will need
+processes for adding maintainers, replacing maintainers, and ensuring
+that new components are maintained.
+
+Becoming a maintainer requires fulfilling the `prerequisites`_
+outlined above.
+
+A new maintainer can be added in any of the following ways:
+
+* An existing maintainer can request that a new person become a
+  component maintainer either in conjunction with the existing
+  maintainer or by replacing them.  If there is no objection from
+  existing maintainers, the role will be conferred.
```

Comment:
I agree. However I think it may make sense that only coding core contributors have voting/veto rights. Thoughts @antoviaque ?

Reply:
@sarina Technical expertise is important to judge the technical capability of a maintainer -- however technical ability is not the only criteria a candidate maintainer has to meet, so even if a core contributor might be unable to judge the technical merits of the candidate, they might have legitimate non-technical objections, about things like social interactions, trust-worthiness, etc.?

Reply:
Ah. I was thinking it would be much like the CC process. My understanding of that process is that only CCs have affirmitive voting rights, but anyone bringing up a concern will be listened to and the nomination process may be halted because of those concerns.

So I misspoke! Would it be reasonable to adopt that model? CCs are the ones who count as "voters" for the +1s, but the community at large is welcome to participate - and will be heard? From OEP-54:


> All community members are welcome to provide comments on the thread.

> CoC violations or other behaviors not upholding the “Conduct” quality of the 3 Cs ought to be raised by anyone, whether in the Discourse thread or to the Program Admins.

> If anyone has concerns they feel cannot be raised publicly, they should [Contact the Program Administrators](https://open-edx-proposals.readthedocs.io/en/latest/processes/oep-0054-core-contributors.html#contact-the-program-administrators) to determine how to proceed. The Administrators should discuss the concerns and try to figure out a way the person could post those concerns themselves; if they cannot, the Administrators should post an anonymized version of the concerns on the Discourse thread.

Reply:
@e0d - I think I would change this a bit. I think _anyone_ should be able to object. If you are saying there's no affirmative process (such as the core contributor 5 votes), I would change this to:

> If there is no objection from existing maintainers, core contributors, or the community, the role will be conferred.

I think it's important that someone in the community be able to bring up a concern - they shouldn't be discounted just because they're not in more of a leadership position.

Reply:
@sarina The [New CC confirmation process](https://github.com/openedx/open-edx-proposals/blob/master/oeps/processes/oep-0054-core-contributors.rst#new-core-contributor-nomination-process) allows anyone to comment, but reserves the right to veto for only existing Core Contributors. I like this model, because it allows anyone to express concern, but protects us from some troll/griefer coming in and voting No on everything just to mess with our process. In the cases where a non-CC's concern is valid, then I'm confident that a CC would support it by issuing a No vote on their behalf.

Tangentially, I also like that the CC nomination process if very specific in where it should be done and how long it should take.

Can we adopt a similar stance for Maintainership? For example:
> * An existing maintainer or core contributor can request that a new person become a component maintainer either in conjunction with the existing  maintainer or by replacing them. The request should be publicized in Discourse, available for all community members to comment their support, questions, and/or concerns. In particular, existing maintainers and core contributors may object to the proposed change. After one week, if there are no objections from existing maintainers or core contributors, the role will be conferred.

Reply:
I don't mind that. I'd like @e0d to give the final signoff on this point.

Reply:
Actually... I've just pushed a change to address this which Ed can review. I don't think this is a controversial change - it lines up maintainership nomination with CC nomination process, and one process is better than two.

https://github.com/openedx/open-edx-proposals/pull/290/commits/26b21cc8f2600192acde8906ecaf1c5367c53b18